models/migrations/v1_21/v260.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package v1_21 //nolint
+
+import (
+	"code.gitea.io/gitea/models/migrations/base"
+
+	"xorm.io/xorm"
+)
+
+func AddLabelsToActRunner(x *xorm.Engine) error {
+	sess := x.NewSession()
+	defer sess.Close()
+
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	type ActionRunner struct {
+		ID           int64
+		AgentLabels  []string
+		CustomLabels []string
+		Labels       []string `xorm:"TEXT"` // new column
+	}
+
+	// add column of `labels` to the `action_runner` table.
+	if err := sess.Sync(new(ActionRunner)); err != nil {
+		return err
+	}
+
+	// combine "agent_labels" col and "custom_labels" col to "labels" col.
+	var runners []*ActionRunner
+	if err := sess.Table("action_runner").Select("id, agent_labels, custom_labels").Find(&runners); err != nil {
+		return err
+	}
+
+	for _, r := range runners {
+		r.Labels = append(r.Labels, r.AgentLabels...)
+		r.Labels = append(r.Labels, r.CustomLabels...)
```

Comment:
I think it should drop the CustomLables, according to the proposal, only agent labels can be used for a runner.

Reply:
That's a good idea. And it means we don't have to combine the data and set a new column, just drop `CustomLabels` column and use `AgentLabels`.

Reply:
Since the `agent_labels` field is deprecated in pb and only `labels` is used, I want to keep it the same in Gitea, so how about `UPDATE action_runner SET labels = agent_labels` and then drop the `agent_labels` and `custom_labels` columns?

Reply:
or use tag mapping?
```
Labels []string `xorm:"agent_labels"`
```

Reply:
I don't really care the name or column, maybe
```go
Labels []string `xorm:"agent_labels"`
```
is enough, to make the migration simple.

Reply:
Maybe we can still call this field as "AgentLabels", only drop the "CustomLables".

Indeed .... the labels are from agent, so AgentLabels is still a right name. And, maybe we would like to have custom labels support again in the future (for display & management purpose only and give them different color/styles to avoid confusing)